homeassistant/components/nissan_leaf/__init__.py

```diff
@@ -154,26 +142,26 @@
                                       utcnow() + INITIAL_UPDATE)
 
     hass.data[DATA_LEAF] = {}
-    tasks = [async_setup_leaf(car) for car in config[DOMAIN]]
-    if tasks:
-        await asyncio.wait(tasks, loop=hass.loop)
+    for car in config[DOMAIN]:
+        setup_leaf(car)
 
-    hass.services.async_register(DOMAIN, SERVICE_UPDATE_LEAF, handle_update,
-                                 schema=UPDATE_LEAF_SCHEMA)
+    hass.services.async_register(
```

Comment:
Use `hass.services.register`.

Reply:
Updated in commit https://github.com/home-assistant/home-assistant/pull/21145/commits/5c5ed5a2b09aac8fcba155ca11d80837ceb86fa2